packages/attune/src/apt/release.rs

```diff
@@ -0,0 +1,147 @@
+                debian_repository_release.codename,
+                debian_repository_release.description
+            FROM
+                debian_repository
+                JOIN debian_repository_release ON debian_repository_release.repository_id = debian_repository.id
+            WHERE
+                debian_repository.tenant_id = $1
+                AND debian_repository.name = $2
+                AND debian_repository_release.distribution = $3
+            LIMIT 1
+            "#,
+            tenant_id.0,
+            repository,
+            distribution,
+        )
+        .fetch_optional(&mut **tx)
+        .await
+        .unwrap()
+    }
+}
+
+#[derive(Debug)]
+pub struct ReleaseFile {
+    pub meta: ReleaseMeta,
+    pub contents: String,
+}
+
+impl ReleaseFile {
+    pub fn from_indexes(
+        release: ReleaseMeta,
+        release_ts: OffsetDateTime,
+        packages_indexes: &Vec<PackagesIndexMeta>,
+    ) -> Self {
+        // Note that the date format is RFC 2822. _Technically_, the Debian spec
+        // says it should be the date format of `date -R -u`, which technically
+        // is RFC 5322, but these formats are compatible. 5322 is a later
+        // revision of 2822 that retains backwards compatibility.
+        let date = release_ts.format(&Rfc2822).unwrap();
+
+        // Prepare "Architectures" and "Components" fields. We use BTreeSets
+        // instead of HashSets to get deterministic iterator order, since index
+        // generation needs to be deterministically replayed.
+        let mut arch_set = BTreeSet::new();
+        let mut comp_set = BTreeSet::new();
+        for p in packages_indexes {
+            arch_set.insert(p.architecture.as_str());
+            comp_set.insert(p.component.as_str());
+        }
+        let archs = arch_set
+            .into_iter()
+            .fold(String::new(), |acc_archs, arch| acc_archs + " " + arch);
+        let archs = archs.strip_prefix(" ").unwrap_or("");
+        let comps = comp_set
+            .into_iter()
+            .fold(String::new(), |acc_comps, comp| acc_comps + " " + comp);
+        let comps = comps.strip_prefix(" ").unwrap_or("");
+
+        // Write release fields.
+        let release_fields: Vec<(&str, Option<String>)> = vec![
+            ("Origin", release.origin.clone()),
+            ("Label", release.label.clone()),
+            ("Version", release.version.clone()),
+            ("Suite", Some(release.suite.clone())),
+            ("Codename", Some(release.codename.clone())),
+            ("Date", Some(date)),
+            ("Architectures", Some(archs.to_string())),
+            ("Components", Some(comps.to_string())),
+            ("Description", release.description.clone()),
+        ];
+        let mut release_file = String::new();
+        for (k, v) in release_fields {
+            if let Some(v) = v {
+                release_file.push_str(&format!("{k}: {v}\n"));
+            }
+        }
```

Comment:
[nit, optional] I'd recommend using functional constructs for stuff like this too; you also get the benefit of not needing to clone!

```suggestion
        let release_file = [
            ("Origin", release.origin.as_ref()),
            ("Label", release.label.as_ref()),
            ("Version", release.version.as_ref()),
            ("Suite", Some(&release.suite)),
            ("Codename", Some(&release.codename)),
            ("Date", Some(&date)),
            ("Architectures", Some(&archs)),
            ("Components", Some(&comps)),
            ("Description", release.description.as_ref()),
        ]
        .fold(String::new(), |mut file, (k, v)| {
            file.push_str(&format!("{k}: {v}\n"));
            file
        });
```

Also FYI, using [`write!`](https://doc.rust-lang.org/std/macro.write.html) is slightly more efficient than `push_str(&format!("..."))` as it can often skip allocations